### PR TITLE
Added Null State Component in NEW-UI

### PR DESCRIPTION
**Pull request by Yash-271120**
Hey! I added the feature where it shows __"No results found for the entered State, Pin code or Date, Please Try Again"__ in the NEW-UI branch. Let me know if there are any issues.!

### CODE REVIEW

**Comment by stephin007** [comment]
Its not working as expected @Yash-271120 
When i select state, it shows the error text even before  i select a district.

Also, 
- i had requested a seperate Component for this.
- What packages did you add? as i see 2 lock files, 
- Please dont add package lock to your commits, as we have already have yarn.lock

**Comment by Yash-271120** [comment]
Ok I will update that. I will create a component named `NullState` which will handle the logic for it!
No other packages where added.
_PS- I am a newbie plz point me where ever I go wrong._

**Comment by stephin007** [comment]
@Yash-271120 any updates regarding this buddy?

**Comment by Yash-271120** [comment]
@stephin007 as I mentioned above I created a new component called `NullState` which takes care of showing the data if available or error message.
You can check the deployed code [here](https://cowin-vaccine-availablity-checker-42k897sqy-stephin007.vercel.app/).
Let me know if it works fine!

**Comment by stephin007** [comment]
@Yash-271120 PEASE RESOLVE THE CONFLICTS so that we can merge your code.

###  Do this by comparing  both the branches, please don't **REMOVE** anything, just add your changes to the file and resolve them!

**Comment by Yash-271120** [comment]
@stephin007 I tried resolving the conflicts. Any further thing giving problem?

**Comment by stephin007** [comment]
@Yash-271120 did you compare the original changes with the NEW-UI branch? did you delete any existing code from the default Branch, if so this change will break the code.

**Comment by stephin007** [comment]
image

Deployment has failed @Yash-271120 , since there are some unused variables

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -326,114 +336,115 @@ const Home = () => {
...
                 />
               </MuiPickersUtilsProvider>
             </div>
           ) : null}
 
-          {vaccineData.length === 0 ? null : (
-            <>
-              <VaccineDataMain vaccineData={currentVaccine} />
-              {pageNumber.length === 1 ? null : (
-                <Pagination
```
Comment (Justinnn07) [inline src/components/Home/Home.js:426]: @Yash-271120 This is required for pagination! please add this!

Thanks

**Comment by Justinnn07** [comment]
image
@Yash-271120 i tried working with your repo, firstly it is not installing the modules secondly please remove your package-lock.json file ..

File: src/components/Home/Home.js (hunk 1 above)
Comment (Yash-271120) [reply src/components/Home/Home.js:426]: @Justinnn07 I again cloned the repo. This time I made the changes and just committed the changed files. No lock files this time.
[New commits](https://github.com/stephin007/Cowin-Vaccine-Availablity-Checker/pull/89/commits/503a6f792269e939fc45258187ac8519b89846b6)
I am new to github so was little bit confused.

**Comment by stephin007** [comment]
@Yash-271120 , can you make null state component UI a bit more attractive rather just showing the Text.


Also, @Justinnn07 also please test all the instances if Null state is working or not

**Comment by Justinnn07** [comment]
> @Yash-271120 , can you make null state component UI a bit more attractive rather just showing the Text.
> 
> Also, @Justinnn07 also please test all the instances if Null state is working or not

sure!

**Review by Justinnn07 — APPROVED** [review]
LGTM! 🔥 @Yash-271120

**Comment by Yash-271120** [comment]
> @Yash-271120 , can you make null state component UI a bit more attractive rather just showing the Text.
@stephin007 sure
Error

Let me know how this UI feels!

**Comment by stephin007** [comment]
@Yash-271120 the image background is not same as that of the app background, it would be good if we remove the white background of the image, using this website : [remove.bg](https://remove.bg)

**Comment by Yash-271120** [comment]
> @Yash-271120 the image background is not same as that of the app background, it would be good if we remove the white background of the image, using this website : [remove.bg](https://remove.bg)
> 
> Also, make sure its responsive in all the mobile devices as well

@stephin007 done!

**Comment by stephin007** [comment]
Can you send a screenshot buddy? @Yash-271120

**Comment by Yash-271120** [comment]
> Can you send a screenshot buddy? @Yash-271120

oie_rp3hRQHbZL4p

**Comment by stephin007** [comment]
@Justinnn07 is this PR good to go?

**Comment by Justinnn07** [comment]
> @Justinnn07 is this PR good to go?
> 
> 

It's not mobile responsive.

**Comment by stephin007** [comment]
@Justinnn07 the latest changes are not yet deployed, ill check again please do review then!

**Comment by stephin007** [comment]
@Justinnn07 please review the code and demo agan, now site is deployed with the latest changes

**Review by Justinnn07 — CHANGES_REQUESTED** [review]
Please remove your package-lock.json

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -92,8 +96,7 @@ const Home = () => {
   const handleDateChange = (date) => {
     setSelectedDate(date);
     setVaccineData([]);
```
[the remark this answers is not in the document]
Comment (stephin007) [reply src/components/Home/Home.js:98]: Why though? @Justinnn07

[the remark this answers is not in the document]
Comment (stephin007) [reply src/components/Home/Home.js:98]: Thats required, because when we are changing the date, we need to clear the list right? thats a feature!

[the remark this answers is not in the document]
Comment (Yash-271120) [reply src/components/Home/Home.js:98]: @Justinnn07 `setVaccineData([])` is required. It stores the fetched data.

**Comment by Justinnn07** [comment]
@Yash-271120 please pull the latest changes from NEW-UI and do the changes over there! 

Also please remove your package-lock.json

Thanks ..

File: src/components/Home/Home.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (stephin007) [reply src/components/Home/Home.js:98]: @Yash-271120 yep yep thats the one, please dont remove it!

One final thing please remove your package lock so that we can finally merge your branch

**Comment by stephin007** [comment]
@Yash-271120 please remove package.lock and we will merge your request!

**Review by Justinnn07 — APPROVED** [review]
LGTM! @Yash-271120 💥🚀

**Comment by stephin007** [comment]
@Justinnn07 wait, deployment has failed 
let me re initiate

**Comment by Justinnn07** [comment]
> @Justinnn07 wait, deployment has failed 
> let me re initiate
> 

I knew that why I didn't merge! Was commenting about this

**Comment by stephin007** [comment]
Awesome work @Yash-271120 , Looking forward for your contribution in other issues!

**Comment by stephin007** [comment]
@Yash-271120 were you committing using 2 accounts 😅 
As i see conversation messages from yash-271120 and commits from @Spidey2711 .


WHy is that so?

**Comment by Yash-271120** [comment]
> @Yash-271120 were you committing using 2 accounts 😅
> As i see conversation messages from yash-271120 and commits from @Spidey2711 .
> 
> WHy is that so?

I also don't know why is that happening. I just have 1 account that's @Yash-271120 .

**Comment by Yash-271120** [comment]
> Awesome work @Yash-271120 , Looking forward for your contribution in other issues!

sure!
also thanks to u guys this was my first PR that got merged :smiley:

**Comment by stephin007** [comment]
Congratulations @Yash-271120 , Shubh Aarambh ❤️

**Comment by stephin007** [comment]
> > @Yash-271120 were you committing using 2 accounts 😅
> > As i see conversation messages from yash-271120 and commits from @Spidey2711 .
> > WHy is that so?
> 
> I also don't know why is that happening. I just have 1 account that's @Yash-271120 . 😕

This is most probably due to incomplete git setup in your local machine. Anyways its okay now!

**Comment by stephin007** [comment]
@Yash-271120 seems like your changes are not responsive , can you fix this as soon as possible!
image

**Comment by Yash-271120** [comment]
> @Yash-271120 seems like your changes are not responsive , can you fix this as soon as possible!
> image

ok i will look into it

**Comment by Yash-271120** [comment]
oie_drAzQR3CzFvH
This look good?
